service/history/commandChecker.go

```diff
@@ -528,6 +528,32 @@ func (v *commandAttrValidator) validateUpsertWorkflowSearchAttributes(
 	return enumspb.WORKFLOW_TASK_FAILED_CAUSE_UNSPECIFIED, nil
 }
 
+func (v *commandAttrValidator) validateModifyWorkflowProperties(
+	namespace namespace.Name,
+	attributes *commandpb.ModifyWorkflowPropertiesCommandAttributes,
+) (enumspb.WorkflowTaskFailedCause, error) {
+	const failedCause = enumspb.WORKFLOW_TASK_FAILED_CAUSE_BAD_MODIFY_WORKFLOW_PROPERTIES_ATTRIBUTES
+	if attributes == nil {
+		return failedCause, serviceerror.NewInvalidArgument(
+			"ModifyWorkflowPropertiesCommandAttributes is not set on command.",
+		)
+	}
+
+	// check at least one attribute is not nil
+	if attributes.UpsertedMemo == nil {
+		return failedCause, serviceerror.NewInvalidArgument(
+			"ModifyWorkflowPropertiesCommandAttributes attributes are all nil.",
+		)
+	}
+
+	// check if UpsertedMemo is not nil, then it's not an empty map
+	if attributes.UpsertedMemo != nil && len(attributes.GetUpsertedMemo().GetFields()) == 0 {
```

Comment:
So, at this moment yes, but the condition above, as new attributes are added, it would become something like `if attributes.UpsertedMemo == nil || attributes.something == nil || ...`. So, this condition here makes sense to check again.